plugins/transport-nio/src/test/java/org/elasticsearch/http/nio/NioHttpServerTransportTests.java

```diff
@@ -309,52 +312,47 @@ public void dispatchBadRequest(final RestRequest request,
-//            transport.start();
-//            final TransportAddress remoteAddress = randomFrom(transport.boundAddress().boundAddresses());
-//
-//            AtomicBoolean channelClosed = new AtomicBoolean(false);
-//
-//            Bootstrap clientBootstrap = new Bootstrap().channel(NioSocketChannel.class).handler(new ChannelInitializer<SocketChannel>() {
-//
-//                @Override
-//                protected void initChannel(SocketChannel ch) {
-//                    ch.pipeline().addLast(new ChannelHandlerAdapter() {});
-//
-//                }
-//            }).group(group);
-//            ChannelFuture connect = clientBootstrap.connect(remoteAddress.address());
-//            connect.channel().closeFuture().addListener(future -> channelClosed.set(true));
-//
-//            assertBusy(() -> assertTrue("Channel should be closed due to read timeout", channelClosed.get()), 5, TimeUnit.SECONDS);
-//
-//        } finally {
-//            group.shutdownGracefully().await();
-//        }
-//    }
+    public void testReadTimeout() throws Exception {
+        final HttpServerTransport.Dispatcher dispatcher = new HttpServerTransport.Dispatcher() {
+
+            @Override
+            public void dispatchRequest(final RestRequest request, final RestChannel channel, final ThreadContext threadContext) {
+                throw new AssertionError("Should not have received a dispatched request");
+            }
+
+            @Override
+            public void dispatchBadRequest(final RestRequest request,
+                                           final RestChannel channel,
+                                           final ThreadContext threadContext,
+                                           final Throwable cause) {
+                throw new AssertionError("Should not have received a dispatched request");
+            }
+
+        };
+
+        Settings settings = Settings.builder()
+            .put(HttpTransportSettings.SETTING_HTTP_READ_TIMEOUT.getKey(), new TimeValue(randomIntBetween(100, 300)))
+            .build();
+
+
+        try (NioHttpServerTransport transport = new NioHttpServerTransport(settings, networkService, bigArrays, pageRecycler,
+            threadPool, xContentRegistry(), dispatcher, new NioGroupFactory(settings, logger))) {
+            transport.start();
+            final TransportAddress remoteAddress = randomFrom(transport.boundAddress().boundAddresses());
+
+            try (NioHttpClient client = new NioHttpClient()) {
+                NioSocketChannel channel = null;
+                try {
+                    AtomicBoolean channelClosed = new AtomicBoolean(false);
+                    channel = client.connect(remoteAddress.address());
+                    channel.addCloseListener((r, t) -> channelClosed.set(true));
+                    assertBusy(() -> assertTrue("Channel should be closed due to read timeout", channelClosed.get()));
```

Comment:
NIT: maybe use a `CountDownLatch` and just `assertTrue("Channel should be closed due to read timeout", channelClosed.await(1L, TimeUnit.MINUTE)));` here instead of the busy assert? (gives a less spammy logging on failure and an easier to interpret thread dump when debugging)?
(Also, I'd actually go for a 1 minute timeout here I think, we adjusted so many network related tests from 10s to 1 min already because we ran into 10s+ GC pauses/freezes in the past)